File: app/src/main/java/com/theappsolutions/boilerplate/util/data/StringUtils.java
package com.theappsolutions.boilerplate.util.data;

/**
 * @author Dmytro Yakovlev d.yakovlev@theappsolutions.com
 * @copyright (c) 2017 TheAppSolutions. (https://theappsolutions.com)
 */
public class StringUtils {

    /**
     * Make only first letter as capitalized and other letter in lower case.
     */
    public static String capitalizeOnlyFirstLetter(String input) {
        return Character.toUpperCase(input.charAt(0)) + input.substring(1).toLowerCase();
    }

    public static boolean hasLetters(String input) {
        return input.matches(".*[^a-z].*");
    }

    public static boolean hasNumbers(String input) {
        return input.matches(".*\\d+.*");
    }

    public static boolean hasUppercase(String input) {
        return input.matches(".*[A-Z].*");
    }
}

File: app/src/main/java/com/theappsolutions/boilerplate/util/data/StringUtils.kt
package com.theappsolutions.boilerplate.util.data

/**
 * @author Dmytro Yakovlev d.yakovlev@theappsolutions.com
 * @copyright (c) 2017 TheAppSolutions. (https://theappsolutions.com)
 */

/**
 * Make only first letter as capitalized and other letter in lower case.
 */
fun capitalizeOnlyFirstLetter(input: String): String {
    return Character.toUpperCase(input[0]) + input.substring(1).toLowerCase()
}

fun hasLetters(input: String): Boolean {
    return input.matches(".*[^a-z].*".toRegex())
}

fun hasNumbers(input: String): Boolean {
    return input.matches(".*\\d+.*".toRegex())
}

fun hasUppercase(input: String): Boolean {
    return input.matches(".*[A-Z].*".toRegex())
}

